server/game/cards/02_SHD/upgrades/VambraceFlamethrower.ts

```diff
@@ -0,0 +1,27 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { UpgradeCard } from '../../../core/card/UpgradeCard';
+import { Location, RelativePlayer, WildcardCardType } from '../../../core/Constants';
+
+export default class VambraceFlamethrower extends UpgradeCard {
+    protected override getImplementationId() {
+        return {
+            id: '6471336466',
+            internalName: 'vambrace-flamethrower',
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.addGainOnAttackAbilityTargetingAttached({
+            title: 'Deal 3 damage divided as you choose among enemy ground units',
+            immediateEffect: AbilityHelper.immediateEffects.distributeDamageAmong({
+                amountToDistribute: 3,
+                canChooseNoTargets: true,
```

Comment:
This is a kind of a niche rules thing, but this should only be true in cases where the card says "any number of units" like overwhelming barrage. In this case picking at least one target is required